src/amp-story-player/amp-story-player-impl.js

```diff
@@ -214,58 +221,33 @@ export class AmpStoryPlayer {
    * @public
    */
   add(stories) {
-    const isStoryObject = (story) =>
-      typeof story === 'object' && story !== null && story.href;
+    const isStoryObject = (story) => {
+      story.iframeIdx = -1;
+      return typeof story === 'object' && story !== null && story.href;
+    };
```

Comment:
Nit:
```suggestion
    const isStoryDef = (story) => story && story.href;
```

Reply:
I think such an inline function should remain pure and not mutate the object you pass to it. This is a type validation function, it's not expected to have side effects.

Also, please rename the function `isStoryDef` to hint at the typedef, and simplify the check as suggested